src/deepsparse/pipeline.py

```diff
@@ -220,8 +228,30 @@ def __call__(self, *args, **kwargs) -> BaseModel:
                 f"Outputs of {self.__class__} must be instances of "
                 f"{self.output_schema} found output of type {type(pipeline_outputs)}"
             )
+        timer.stop("post_process")
+        timer.stop("total_inference")
+
+        if not monitoring:
+            return pipeline_outputs
+
+        else:
+            data = None
+            timing = timer.build()
+            return pipeline_outputs, timing, data
```

Comment:
Thoughts on storing the timing as `self._last_timing = timer.build`, and then maybe adding a method for `def get_times_for_last_run(self) -> Optional[InferenceTimes]`?

Then we wouldn't need the additional flag in this method, but still have a clear way to access it.

Reply:
Oooh yeah that's a great call out about reliably returning the last timings and the data stuff.

Will the PipelineLogger class be outside of Pipeline? Or will all this timing data be passed directly to PipelineLogger from inside Pipeline (maybe PipelineLogger is passing into constructor of Pipeline)?

Reply:
yup, just want to call out here that we should never keep state in a pipeline that relates to a single inference run (esp the latest) as this will lead to race conditions in async runs